File: IRemember/app/src/main/java/ru/goodibunakov/iremember/PreferenceHelper.java
package ru.goodibunakov.iremember;

import android.content.Context;
import android.content.SharedPreferences;

/**
 * Created by GooDi on 16.09.2017.
 */

class PreferenceHelper {

    private static PreferenceHelper instance;
    static final String SPLASH_IS_INVISIBLE = "splash_is_invisible";
//    private Context context;
    private SharedPreferences preferences;

    private PreferenceHelper() {
    }

    static PreferenceHelper getInstance() {
        if (instance == null) {
            instance = new PreferenceHelper();
        }
        return instance;
    }

    void init(Context context) {
//        this.context = context;
        preferences = context.getSharedPreferences("preferences", Context.MODE_PRIVATE);
    }

    void putBoolean(String key, boolean value) {
        SharedPreferences.Editor editor = preferences.edit();
        editor.putBoolean(key, value);
        editor.apply();
    }

    Boolean getBoolean(String key) {
        return preferences.getBoolean(key, false);
    }
}

File: IRemember/app/src/main/java/ru/goodibunakov/iremember/utils/PreferenceHelper.kt
package ru.goodibunakov.iremember.utils

import android.content.Context
import android.content.SharedPreferences

class PreferenceHelper {

    private var preferences: SharedPreferences? = null

    companion object {

        const val SPLASH_IS_INVISIBLE = "splash_is_invisible"

        private var instance: PreferenceHelper? = null

        fun getInstance(): PreferenceHelper {
            if (instance == null) {
                instance = PreferenceHelper()
            }
            return instance!!
        }
    }

    fun init(context: Context) {
        preferences = context.getSharedPreferences("preferences", Context.MODE_PRIVATE)
    }

    fun putBoolean(key: String, value: Boolean) {
        val editor = preferences!!.edit()
        editor.putBoolean(key, value)
        editor.apply()
    }

    fun getBoolean(key: String): Boolean {
        return preferences!!.getBoolean(key, false)
    }
}